### PR TITLE
Updates and fixes for the documentation

### PR DESCRIPTION
**Pull request by froschdesign**
* Updates MkDocs configuration
* Fixes headline orders
* Removes old HTML index page
* Removes some PHP tags (only needed in code examples for templates)
* Removes an old note for a to-do

### CODE REVIEW

File: mkdocs.yml [hunk 1]
```diff
@@ -20,6 +20,5 @@ pages:
         - "Using AoT with Expressive and zend-servicemanager": cookbook/aot-guide.md
     - "Migration Guide": migration.md
 site_name: zend-di
-site_description: zend-di
+site_description: "Automated dependency injection and instance manager."
```
Comment (tux-rampage) [inline mkdocs.yml]: There is no instance manager in v3 anymore, in favor of psr containers and ServiceManager.

Comment (froschdesign) [reply mkdocs.yml]: The description was taken from the [component list](https://zendframework.github.io/zf-mkdoc-theme/scripts/zf-component-list.json) which is used for the component overview on [docs.zendframework.com](https://docs.zendframework.com/).

Do you have an idea to renew the description? (The description in the readme file of this repository is too long.)

Comment (tux-rampage) [reply mkdocs.yml]: Hmm, how about the following descriptions?

> Automated dependency injection for PSR-11 containers

A less generic one but more ZF specific could be:

> Automated dependency injection for ZendServicemanager

Thoughts? I guess the component list and composer.json description should be updated then as well.

Comment (froschdesign) [reply mkdocs.yml]: > Automated dependency injection for PSR-11 containers

👍 

> I guess the component list and composer.json…

Right! And also the repository description.

Comment (tux-rampage) [reply mkdocs.yml]: > And also the repository description.

I can't do that. Can you? Otherwise @Ocramius / @weierophinney may jump in.

Comment (froschdesign) [reply mkdocs.yml]: @weierophinney and @ezimuel

**Pull request merged**